frame/broker/src/dispatchable_impls.rs

```diff
@@ -0,0 +1,416 @@
+						Some(AllowedRenewalRecord { completion: Partial(w), price: p })
+							if price == p =>
+							w,
+						_ => CoreMask::void(),
+					} | region_id.mask;
+					let workload =
+						if assigned.is_complete() { Complete(workplan) } else { Partial(assigned) };
+					let record = AllowedRenewalRecord { price, completion: workload };
+					AllowedRenewals::<T>::insert(&renewal_id, &record);
+					if let Some(workload) = record.completion.drain_complete() {
+						Self::deposit_event(Event::Renewable {
+							core: region_id.core,
+							price,
+							begin: region.end,
+							workload,
+						});
+					}
+				}
+			}
+			Self::deposit_event(Event::Assigned { region_id, task: target, duration });
+		}
+		Ok(())
+	}
+
+	pub(crate) fn do_pool(
+		region_id: RegionId,
+		maybe_check_owner: Option<T::AccountId>,
+		payee: T::AccountId,
+		finality: Finality,
+	) -> Result<(), Error<T>> {
+		if let Some((region_id, region)) = Self::utilize(region_id, maybe_check_owner, finality)? {
+			let workplan_key = (region_id.begin, region_id.core);
+			let mut workplan = Workplan::<T>::get(&workplan_key).unwrap_or_default();
+			let duration = region.end.saturating_sub(region_id.begin);
+			if workplan
+				.try_push(ScheduleItem { mask: region_id.mask, assignment: CoreAssignment::Pool })
+				.is_ok()
+			{
+				Workplan::<T>::insert(&workplan_key, &workplan);
+				let size = region_id.mask.count_ones() as i32;
+				InstaPoolIo::<T>::mutate(region_id.begin, |a| a.private.saturating_accrue(size));
+				InstaPoolIo::<T>::mutate(region.end, |a| a.private.saturating_reduce(size));
+				let record = ContributionRecord { length: duration, payee };
+				InstaPoolContribution::<T>::insert(&region_id, record);
+			}
+
+			Self::deposit_event(Event::Pooled { region_id, duration });
+		}
+		Ok(())
+	}
+
+	pub(crate) fn do_claim_revenue(
+		mut region: RegionId,
+		max_timeslices: Timeslice,
+	) -> DispatchResult {
+		let mut contribution =
+			InstaPoolContribution::<T>::take(region).ok_or(Error::<T>::UnknownContribution)?;
+		let contributed_parts = region.mask.count_ones();
+
+		Self::deposit_event(Event::RevenueClaimBegun { region, max_timeslices });
+
+		let mut payout = BalanceOf::<T>::zero();
+		let last = region.begin + contribution.length.min(max_timeslices);
+		for r in region.begin..last {
+			region.begin = r + 1;
+			contribution.length.saturating_dec();
+
+			let mut pool_record = match InstaPoolHistory::<T>::get(r) {
+				Some(x) => x,
+				None => continue,
+			};
+			let total_payout = match pool_record.maybe_payout {
+				Some(x) => x,
+				None => break,
+			};
+			let p = total_payout
+				.saturating_mul(contributed_parts.into())
+				.checked_div(&pool_record.private_contributions.into())
+				.unwrap_or_default();
+
+			payout.saturating_accrue(p);
+			pool_record.private_contributions.saturating_reduce(contributed_parts);
+
+			let remaining_payout = total_payout.saturating_sub(payout);
```

Comment:
`payout` is a total payout of the current claim, probably `remaining_payout` is `total_payout - p`